app/controllers/favorites_controller.rb

```diff
@@ -1,7 +1,41 @@
 class FavoritesController < ApplicationController
 
   def index
+    @favorite_pets = @favorite.favorite_pets
+  end
+
+  def update
+    pet = Pet.find(params[:pet_id])
+    @favorite = Favorite.new(session[:favorite])
+    if session[:favorite].nil? || session[:favorite][pet.id.to_s].nil?
+      @favorite.add_pet(pet.id)
+      session[:favorite] = @favorite.contents
+      quantity = @favorite.count_of(pet.id)
+      flash[:notice] = "You have added #{pet.name} to your favorites"
+    end
```

Comment:
This is nearly coppied from the `Cart & Order Life Cycle` lesson on thursday

Reply:
![](https://media.giphy.com/media/Ie9biTPXxhmhKn7cvX/giphy.gif)